src/libra_py/cube_file_process.py

```diff
@@ -0,0 +1,255 @@
+"""
+.. module:: cube_file_process
+   :platform: Unix, Windows
+   :synopsis: This module implements functions for processing with cube files
+.. moduleauthors:: 
+       Mohammad Shakiba, Brendan Smith, Alexey V. Akimov 
+  
+"""
+
+
+import os
+import sys
+import math
+import re
+
+if sys.platform=="cygwin":
+    from cyglibra_core import *
+elif sys.platform=="linux" or sys.platform=="linux2":
+    from liblibra_core import *
+
+import numpy as np
+
+
+
+
+def read_cube(filename: str):
+    """
+    This function reads the wavefunction from a cube file and stores it in
+    a 1D numpy array
+
+    Args:
+
+        filename ( string ): the name of the .cube file to read
+
+    Returns:
+
+        numpy.array : the 1D array of the wavefunctions for all the points on the grid
+    
+
+    Note: 
+
+        Previously, it was called `read_1`
+
+    """
+    
+    f = open(filename,'r')
+    lines = f.readlines()
+    f.close()
+
+    natoms = int(lines[2].split()[0])
+    # We skip a few lines in the cube files, that go as follows:
+    # 2 lines - comments
+    # 1 line  - the number of atoms, etc.
+    # 3 lines - the grid spacing and number of grid points in each dimensions
+
+    nstart = natoms+2+1+3        # the index of the first line containing wfc data
+    nlines = len(lines)          # the total number of lines
+
+    x = []
+    for i in range(nstart,nlines):
+        tmp = lines[i].split()
+        ncols = len(tmp)
+
+        for j in range(ncols):
+            x.append(float(tmp[j]))
+    
+    data = np.array(x)
+    #data = np.loadtxt(filename,skiprows=n)
+    
+    return data
+
+
+
+
+def grid_volume(filename: str):
+    """
+	This function reads the wavefunction from a cube file and calculate 
+	the grid volum using the X-, Y- and Z-axis of the volumetric region
+	which are placed in the 4th, 5th and 6th line of the cube file structure
+	
+        Args:
+        
+	    filename (string): The name of the .cube file to read.
+		
+        Returns:
+	
+	    dv (float): The grid volume in Bohr^3.
+		
+        Note:
+        
+	    It uses the formula of calculating the parallelpiped volume:
+            https://en.wikipedia.org/wiki/Parallelepiped
+        	
+    """
+	
+    f = open(filename,'r')
+    lines = f.readlines()
+    f.close()
+    
+    # We use the 3rd, 4th and 5th row in the lines to obtain
+    # the vectors of the parallelpiped into a numpy array (V).
+    V = np.array([[float(lines[3].split()[1]),float(lines[3].split()[2]),float(lines[3].split()[3])],\
+                 [float(lines[4].split()[1]),float(lines[4].split()[2]),float(lines[4].split()[3])],\
+                 [float(lines[5].split()[1]),float(lines[5].split()[2]),float(lines[5].split()[3])]\
+                 ])
+    # Then we calculate the determinant of the a to obtain the volume.
+    dv = np.absolute(np.linalg.det(V))
+	
+    
+    return dv 
+
+
+
+
+def read_volumetric_data(filename: str):

```

Comment:
this function does too many things - consider breaking it down into several more modular units